### PR TITLE
Create event api

### PR DESCRIPTION
**Pull request by prakashchoudhary07**
#34

### CODE REVIEW

File: controllers/events.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        name: eventData.name,
+        description: eventData.description,
+        usersId: userData?.id,
+        location: eventData.location,
+        EventType: eventData.eventType,
+        parentEventID: parentEvent.id,
+      },
+    });
+
+    console.log(event);
```
Comment (ankurnarkhede) [inline controllers/events.ts:47]: Add log to identify the specific log.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

**Comment by prakashchoudhary07** [comment]
In Doc, we are passing calendarId for POST data, but we have no field to store calendarId in parentEvent or childEvent.
I belive we will be adding that in parentEvent right? Correct me if I am missing some thing here

@ankurnarkhede

**Comment by ankurnarkhede** [comment]
> In Doc, we are passing calendarId for POST data, but we have no field to store calendarId in parentEvent or childEvent. I belive we will be adding that in parentEvent right? Correct me if I am missing some thing here
> 
> @ankurnarkhede

We need to add it in parentEvent schema. Its already added here: https://github.com/Real-Dev-Squad/calendar-API/blob/512d797cbb065ad06bf6123b41859b1eacaef48d/prisma/schema.prisma#L65-L66

**Comment by prakashchoudhary07** [comment]
> 

Will we be storing calendarId in eventTypeId?

**Comment by ankurnarkhede** [comment]
> > 
> 
> Will we be storing calendarId in eventTypeId?

No, eventTypeId will be the FK from EventType table. It has nothing to do with calendarId.

File: constants/recurringFrequency.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export const RECURRING_FREQUENCY: any = {
```
Comment (ankurnarkhede) [inline constants/recurringFrequency.ts:1]: Lets add all config to config object itself.

File: controllers/events.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import { ChildEvent, EventType } from "@prisma/client";
+import { Request, Response } from "express";
+import Boom from "@hapi/boom";
+import prisma from "../prisma/prisma";
+import {
+  createManyChildEvent,
+  findParentEvent,
+} from "../services/eventsService";
+import { formateParentEvent } from "../utils/formate-response";
```
Comment (ankurnarkhede) [inline controllers/events.ts:9]: format*

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]
Still reviewing, will add the full review in sometime.

**Comment by harshith-venkatesh** [comment]
* Recurring logic is currently applied for daily frequency

File: controllers/events.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    const attendeesList = eventData.attendees ?? [];
+
+    // Get event id from event name
+    const eventTypeData: EventType = await prisma.eventType.findFirstOrThrow({
+      where: {
+        name: eventData.eventType,
+      },
+    });
+
+    // Get attnedee ids from email
```
Comment (iRohitGaur) [inline controllers/events.ts:29]: `attnedee` --> `attendee`

File: routes/events.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Router } from "express";
+import { postEvent } from "../controllers/events";
+import authenticate from "../middlewares/authenticate";
+import { validate } from "../middlewares/validators/validator";
+import { postEventSchema } from "../middlewares/validators/zod-schemas/events";
+
+const router = Router();
+
+// eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (iRohitGaur) [inline routes/events.ts:9]: why disable?

File: routes/events.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+router.post("/", authenticate, validate(postEventSchema), postEvent);
```
Comment (iRohitGaur) [inline routes/events.ts:10]: why disable?

File: services/eventsService.ts [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+const DATE_ADD_MAPPING: any = {
+  DAILY: addDays,
+  WEEKLY: addWeeks,
+  MONTHLY: addMonths,
+  YEARLY: addYears,
+};
+Object.freeze(DATE_ADD_MAPPING);
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/explicit-function-return-type
+// const handelDailyEventsTimeUpdate = () => {};
```
Comment (iRohitGaur) [inline services/eventsService.ts:13]: suggestion: should be removed if not required.

**Review by DeRaowl — APPROVED** [review]

**Pull request merged**